tests/test_datasets.py

```diff
@@ -63,9 +63,10 @@ def test_load_moscow_stjoes_as_xy_as_frame():
 
 def test_asframe_raises_without_pandas():
     """Test that as_frame=True raises a helpful error if pandas is not installed."""
-    with mock.patch.dict(sys.modules, {"pandas": None}):
-        with pytest.raises(ImportError, match="pip install pandas"):
-            load_moscow_stjoes(as_frame=True)
+    with mock.patch.dict(sys.modules, {"pandas": None}), pytest.raises(
+        ImportError, match="pip install pandas"
+    ):
+        load_moscow_stjoes(as_frame=True)
```

Comment:
Whoa! I've never seen that syntax before on a `with` statement (multiple statements).  I assume they happen sequentially.

Reply:
That was new to me too! Recommended by [SIM117](https://beta.ruff.rs/docs/rules/multiple-with-statements/). From [the Python docs](https://docs.python.org/3/reference/compound_stmts.html#with):

![image](https://github.com/lemma-osu/scikit-learn-knn-regression/assets/50475791/cf11e5ff-d73e-4180-bd6c-7af9da439145)